client/client.go

```diff
@@ -281,21 +281,97 @@ func newHttpClient() *http.Client {
 	return &http.Client{}
 }
 
+func loadServerTrust(ctx *openssl.Ctx, conf *Config) (*openssl.Ctx, error) {
+	// Trust CA's in the standard location (/etc/ssl/certs), and the
+	// configured server certificate when building the certificate chain
+	err := ctx.LoadVerifyLocations(conf.ServerCert, "/etc/ssl/certs/")
+	return ctx, err
+}
+
+func loadClientTrust(ctx *openssl.Ctx, conf *Config) (*openssl.Ctx, error) {
```

Comment:
we should call here `SSL_CTX_clear_chain_certs` just to be safe (technically we do not know where did ctx come from), but the wrapper currently does not have this function, I will add it to the wrapper, and then come up with a PR.

Reply:
Could you expand a little here, I'm not following. We do load the context ourselves, and it is a new ctx everytime?

Reply:
yes, it is new every time. cleaning the chain is just a safety precaution. `loadClientTrust` is expected to load the trust, and it does not know anything about previous whereabouts of ctx. not a big deal, and not a blocker here.